packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+///
+/// If shortcuts should be generally enabled, but are not easily defined in the
+/// context surrounding the menu bar, consider using a [ShortcutRegistry] to
+/// register them. To be sure that selecting a menu item and triggering the
+/// shortcut do the same thing, it is recommended that they trigger the same
+/// [Intent] or call the same callback.
+/// {@endtemplate}
+///
+/// Selecting a menu item causes the [MenuItemButton.onSelected] callback to be
+/// called, or the [MenuItemButton.onSelectedIntent] intent to be fired,
+/// depending on which is set.
+///
+/// When a menu item with a submenu is clicked on, it toggles the visibility of
+/// the submenu. When the menu item is hovered over, the submenu will open, and
+/// hovering over other items will close the previous menu and open the newly
+/// hovered one. When those occur, [MenuButton.onOpen], and
+/// [MenuButton.onClose] are called on the corresponding [MenuButton] child of
+/// the menu bar.
+///
+/// {@tool dartpad}
+/// This example shows a [MenuBar] that contains a single top level menu,
+/// containing three items for "About", a checkbox menu item for showing a
+/// message, and "Quit". The items are identified with an enum value.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_bar.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [MenuButton], a menu item which manages a submenu.
+/// * [MenuItemGroup], a menu item which collects its members into a group
+///   separated from other menu items by a divider.
+/// * [MenuItemButton], a leaf menu item which displays the label, an optional
+///   shortcut label, and optional leading and trailing icons.
+/// * [createCascadingMenu], a function that creates a [MenuEntry] that allows
+///   creation and management of a cascading menu anywhere.
+/// * [MenuController], a class that allows controlling and connecting menus.
+/// * [PlatformMenuBar], which creates a menu bar that is rendered by the host
+///   platform instead of by Flutter (on macOS, for example).
+/// * [ShortcutRegistry], a registry of shortcuts that apply for the entire
+///   application.
+/// * [VoidCallbackIntent] to define intents that will call a [VoidCallback] and
+///   work with the [Actions] and [Shortcuts] system.
+/// * [CallbackShortcuts] to define shortcuts that simply call a callback and
+///   don't involve using [Actions].
+class MenuBar extends StatefulWidget with DiagnosticableTreeMixin {
+  /// Creates a const [MenuBar].
+  const MenuBar({
+    super.key,
+    this.controller,
+    this.backgroundColor,
+    this.minimumHeight,
+    this.padding,
+    this.elevation,
+    this.shape,
+    this.expand = true,
+    this.children = const <Widget>[],
+  });
+
+  /// An optional controller that allows outside control of the menu bar.
+  ///
+  /// You can use a controller to close any open menus from outside of the menu
+  /// bar using [MenuController.closeAll].
```

Comment:
Should we remind the reader that if they provide a controller, they're responsible for disposing it?